### DISCUSSION TITLE
Any way for renderer to return something other than string?

### DISCUSSION THREAD

**Discussion by nrivard** [markedjs/marked#3349] [General]
I know this library is pretty HTML specific but my project is using defined JSX types that I want to return rather than a string (I have to do further processing on the objects themselves before sending out as JSON). Ex:

```ts
heading({ text, depth }): AnyComponent {
  return <Title title={text} font={`heading_${depth}`} />
}
```
Is it possible to instead have a generic renderer/parser?

```ts
declare class Parser<Output> {
  renderer: Renderer<Output>;

  parse(tokens: Token[], top?: boolean): Output;
}

declare class Renderer<Output> {
  heading(token: Tokens.Heading): Output;
}
```

**Comment by UziTech** [comment]
No because of the way the renderers use the parser. It is possible for you to only use the Lexer then create your own parser and renderers to parse the tokens to jsx. I believe that is what https://www.npmjs.com/package/marked-react does.
